TODO.md

```diff
@@ -0,0 +1,69 @@
+# TODO
+remaining tasks for the `0.2` release, in order of priority
+
+## major
+* [ ] move behaviours (getting, updating, etc.) to library methods
+* [ ] validation (schema, column, relationship)
+* [ ] declarative entity definitions
+* [ ] schema validation
+* [ ] documentation
+* [ ] callbacks
+* [ ] pagination
+* [ ] hateoas-style support
+
+### move behaviours
+at the moment, when you generate your code, a lot of functionality
+gets put in the restplus endpoint. this results in huge diffs between
+versions and is tiresome to test.
+
+it is proposed that this code moves into a library function which then
+gets imported by the endpoint. this will make the code easier to test,
+and will result in a smaller generated codebase.
+
+### validation
+there's loads that could be done to validate a schema. the most interesting
+part is probably the relationships - to do this we would need a two-pass
+implementation where the first pass builds up an idea of what entities are
+present and how they relate to one another, and the second pass
+verifies that these relationships are possible.
+
+error reporting should be as approachable and helpful as possible. see
+[elm](https://elm-lang.org/) for inspiration.
+
+### declarative entity definitions
+writing entities declaratively is much more natural to python developers
+already familiar with sqlalchemy, which is a dependency of genyrator.
+it is proposed to use decorators in the style of
+[attr.s](http://www.attrs.org/en/stable/) to configure the behaviour of
+entities.
+
+### schema validation
+we currently validate provided data with a really old version
+of marhsmallow-sqlalchemy. pros: it gives nice errors and will persist
+the data if it's valid. cons: it's not easily configurable from the standpoint
+of someone writing an entity. explore going with a different library, writing our
+own or something different? there could be the opportunity to allow the option
+to specify column-level validation - how would this look?
+
+
+### callbacks
```

Comment:
I think it would be interesting to step back and think of different ways of approaching this. Are callbacks the right answer or being able to fully override certain aspects. What are the actual user needs that we have come across? What would be most natural to a developer that already knows flask-restplus / flask / sqlalchemy / marshmallow ...?

Reply:
this is a fair point, I had this in here more as a place to stick ideas. the idea was that if you just want to customise a specific endpoint, or do a little more processing on all `GET`s or something like that, you could sneak your own code in.